src/shims/io.rs

```diff
@@ -48,21 +48,16 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             .memory()
             .read_c_str(this.read_scalar(path_op)?.not_undef()?)?;
         let path = std::str::from_utf8(path_bytes)
-            .map_err(|_| err_unsup_format!("{:?} is not a valid utf-8 string", path_bytes))?;
-
-        match File::open(path) {
-            Ok(file) => {
-                let mut fh = &mut this.machine.file_handler;
-                fh.low += 1;
-                fh.handles.insert(fh.low, FileHandle{ file, flag});
-                Ok(fh.low)
-            }
-
-            Err(e) => {
-                this.machine.last_error = e.raw_os_error().unwrap() as u32;
-                Ok(-1)
-            }
-        }
+            .map_err(|_| err_unsup_format!("{:?} is not a valid utf-8 string", path_bytes))?
+            .to_owned();
```

Comment:
ŵhy the `to_owned`?

Reply:
Probably it was some residue of a borrowing issue I had earlier. Will fix